src/util/collisionhelper.cpp

```diff
@@ -0,0 +1,75 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2021                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <openspace/util/collisionhelper.h>
+
+namespace openspace::collision {
+
+// Source: http://paulbourke.net/geometry/circlesphere/raysphere.c
+bool lineSphereIntersection(glm::dvec3 p1, glm::dvec3 p2, glm::dvec3 center,
+                            double r, glm::dvec3& intersectionPoint)
+{
+    long double a, b, c;
```

Comment:
Can move the variable declarations further down where they are actually used to make it less old-school C style

Reply:
Are the `long double` really necessary?  Long double is not standardized and might produce some problems.  Maybe `double` is enough?

Reply:
You're probably right that a double is enough. In the worst case the diffs below would be in the order of the size of the universe, i.e. 10^27. We want to square those values and sum them together, but it still shouldn't result in larger numbers than the order of 10^54 something, if I'm thinking correctly. So a double is still more than enough

Reply:
Note to self: The same goes for all the functions in this file